File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {    
    people.sort((a, b) => a - b);
    let startIdx = 0;
    let lastIdx = people.length - 1;
    let boat = 0;

    // 30 50 50 70 100 [limit : 100]
    // [100]
    // [30, 70] lastIdx = N - 2
    // [50, 50] lastIdx = N - 3
    while (startIdx <= lastIdx) {
        if (people[startIdx] + people[lastIdx] <= limit) {
            startIdx++;
        }
        lastIdx--;
        boat++;
    }
    
    return boat;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    cnt = 0
    lIdx = 0
    rIdx = len(people) - 1
    
    people.sort()
    
    while lIdx <= rIdx:
        if people[lIdx] + people[rIdx] <= limit:
            lIdx += 1
        rIdx -= 1
        cnt += 1
        
    return cnt